### ISSUE TITLE
[BUG] False positive: spancheck flags wrapped tracer.Start as unassigned span

### ISSUE THREAD

**Issue by jonas0616** [jjti/go-spancheck#33]
**Describe the bug**
`spancheck` currently reports a false positive `span is unassigned, probable memory leak (spancheck)` when using a wrapper function that returns the result of `tracer.Start`. Even when the returned `Span` is assigned to a variable and properly ended with `span.End()`.

**To Reproduce**
```go
func startSpan(ctx context.Context, spanName string) (context.Context, oteltrace.Span) {
    tracer := otel.GetTracerProvider().Tracer(ScopeName)
    return tracer.Start(ctx, spanName, oteltrace.WithSpanKind(oteltrace.SpanKindInternal))
}
```

**Expected behavior**
Calls to `tracer.Start` that are wrapped in helper functions and whose Span result is assigned and ended properly by the caller should not trigger a warning.

**Comment by jjti** [comment]
@jonas0616 thank you for reaching out! Could you check out this https://github.com/jjti/go-spancheck?tab=readme-ov-file#ignore-check-signatures setting? It's like below in golangcilint:

```yaml
linters-settings:
  spancheck:
    # A list of regexes for function signatures that silence `record-error` and `set-status` reports
    # if found in the call path to a returned error.
    # https://github.com/jjti/go-spancheck#ignore-check-signatures
    # Default: []
    ignore-check-signatures:
      - "mypkg.startSpan"
```

If you want to link for any spans that are returned from the `startSpan` function, you'd instead use the `extra-start-span-signatures` setting: https://github.com/jjti/go-spancheck?tab=readme-ov-file#extra-start-span-signatures

**Comment by jonas0616** [comment]
Thanks @jjti , that makes sense! I’ll try setting extra-start-span-signatures for my startSpan function. Appreciate the clarification!
